src/fparser/two/utils.py

```diff
@@ -535,7 +539,8 @@ def match(startcls, subclasses, endcls, reader,
                 found_end = True
                 break
             # Return to start of classes list now that we've matched
-            i = 0
+            if not strict_order:
+                i = 0
```

Comment:
``i`` is special cased in pylint as a reasonable counting variable, so doesn't show up (as far as I can see - but there is a *lot* of noise from this file). To be honest, I gave up with pylint about 6 years ago - pycodestyle has definitely won the mindshare in the scientific python sphere FWICT. If you're keen to have it run on fparser code then I'll dust it off and make use of it more when developing for it.

PEP8's perspective on ``i`` as a variable name is fairly non-committal... looks like most single character variable names are ["reasonable"](https://www.python.org/dev/peps/pep-0008/#names-to-avoid).

Reply:
You're quite right about i as a variable - pylint is happy with it. That is new to me. It doesn't like z though, so i is certainly special :-).

I think we will stick with pycodestyle and pylint conformance for PSyclone and fparser. We've been reasonably successful with these for PSyclone and are slowly getting there with fparser, but it will take a while as we are dealing with existing non-conformant code. Note, conformance to pylint does not need to be 10/10. Essentially, if the rule is unreasonable in the circumstances then we do not enforce it. Where it is agreed to be unreasonable, we allow pylint directives to be added into the code to disable the particular warning.